libraries/WiFi/src/WiFiProv.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+    WiFiProv.cpp - WiFiProv class for provisioning
+    All rights reserved.
+ 
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) any later version.
+  
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+  
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+    
+*/
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <string.h>
+#include <esp_err.h>
+#include <esp_wifi.h>
+#include <esp_event_loop.h>
+#include <esp32-hal.h>
+
+#include <nvs_flash.h>
+#include <wifi_provisioning/scheme_ble.h>
+#include <wifi_provisioning/scheme_softap.h>
+#include <wifi_provisioning/manager.h>
+
+#include "WiFiProv.h"
+
+static const char *TAG = "Provision_in_Arduino";
+uint8_t custom_service_uuid[16] = {  0xb4, 0xdf, 0x5a, 0x1c, 0x3f, 0x6b, 0xf4, 0xbf,
+                                     0xea, 0x4a, 0x82, 0x03, 0x04, 0x90, 0x1a, 0x02, };
+#define SERV_NAME_PREFIX_BLE "BLE_"
+#define SERV_NAME_PREFIX_WIFI "WIFI_"
+esp_err_t (*WiFiEventCb)(void *ctx,system_event_t *event) = NULL;
+
+     
+static esp_err_t event_handler(void *ctx, system_event_t *event)
+{
+    
+    wifi_prov_mgr_event_handler(ctx, event);
+    WiFiEventCb(ctx,event);
+    return ESP_OK;
+}
+
+static void wifi_init_sta()
+{
+    ESP_ERROR_CHECK(esp_wifi_set_mode(WIFI_MODE_STA) );
+    log_i("\nAleardy Provisioned, starting Wi-Fi STA\n");
```

Comment:
why the extra new lines? this is one before and two after (log_x adds it's own new line as well)